underfs/glusterfs/src/main/java/alluxio/underfs/glusterfs/GlusterFSUnderFileSystem.java

```diff
@@ -27,45 +31,68 @@
 
   /**
    * Constant for the Gluster FS URI scheme.
    */
   public static final String SCHEME = "glusterfs://";
 
   /**
-   * Constructs a new Gluster FS {@link UnderFileSystem}.
+   * Prepares the Hadoop configuration necessary.
    *
-   * @param uri the {@link AlluxioURI} for this UFS
-   * @param conf the configuration for Hadoop or GlusterFS
+   * @param path the gluster path to serve as ufs
+   * @param ufsConf Hadoop configuration
+   * @return a prepared configuration
    */
-  public GlusterFSUnderFileSystem(AlluxioURI uri, Object conf) {
-    super(uri, conf);
-  }
-
-  @Override
-  public String getUnderFSType() {
-    return "glusterfs";
-  }
-
-  @Override
-  protected void prepareConfiguration(String path,
-      org.apache.hadoop.conf.Configuration hadoopConf) {
+  public static Configuration prepareConfiguration(String path, Map<String, String> ufsConf) {
     if (path.startsWith(SCHEME)) {
+      Configuration hadoopConf = new Configuration();
       // Configure for Gluster FS
-      hadoopConf.set("fs.glusterfs.impl", Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_IMPL));
-      hadoopConf.set("mapred.system.dir", Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_MR_DIR));
-      hadoopConf
-          .set("fs.glusterfs.volumes", Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES));
-      hadoopConf.set(
-          "fs.glusterfs.volume.fuse." + Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES),
-          Configuration.get(PropertyKey.UNDERFS_GLUSTERFS_MOUNTS));
+      hadoopConf.set("fs.glusterfs.impl",
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_IMPL, ufsConf));
+      hadoopConf.set("mapred.system.dir",
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_MR_DIR, ufsConf));
+      hadoopConf.set("fs.glusterfs.volumes",
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES, ufsConf));
+      hadoopConf.set("fs.glusterfs.volume.fuse."
+          + UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_VOLUMES, ufsConf),
+          UnderFileSystemUtils.getValue(PropertyKey.UNDERFS_GLUSTERFS_MOUNTS, ufsConf));
+      return hadoopConf;
     } else {
       // If not Gluster FS fall back to default HDFS behavior
       // This should only happen if someone creates an instance of this directly rather than via the
       // registry and factory which enforces the GlusterFS prefix being present.
-      super.prepareConfiguration(path, hadoopConf);
+      return HdfsUnderFileSystem.createConfiguration(ufsConf);
     }
   }
+
+  /**
+   * Factory method to construct a new Gluster FS {@link UnderFileSystem}.
+   *
+   * @param uri the {@link AlluxioURI} for this UFS
+   * @param conf the configuration for Hadoop or GlusterFS
+   * @return a new Gluster FS {@link UnderFileSystem}
+   */
+  public static GlusterFSUnderFileSystem createInstance(AlluxioURI uri, Map<String, String> conf) {
```

Comment:
Let me know if you want to change `createInstance` across different ufs adaptors, if yes, I could change them in this PR.